app/controllers/application_controller.rb

```diff
@@ -2122,9 +2122,9 @@ def set_global_session_data
     session[:imports] = @sb[:imports] ? @sb[:imports] : nil # Imported file data from 2 stage import
 
     # Save @edit and @view in session, if present
-    if @lastaction == "show_list"                           # If show_list was the last screen presented or tree is being autoloaded save @edit
-      @edit ||= session[:edit]                              #   Remember the previous @edit
-      @view ||= session[:view]                              #   Remember the previous @view
+    if @lastaction == "show_list" || controller_name == "picture" # If show_list was the last screen presented or tree is being autoloaded save @edit
```

Comment:
@hstastna this change is not required, controller_name will never be "picture", you should remove this commit.

Reply:
Generally, this change does not make sense, but I haven't found other solution yet, to make Advanced search work. Without this change, when opening Adv search, there is an error and Adv search stays open and everything stops working there. Anyway, I will continue to investigate why I get controller name "picture" there and will look for better solution. Maybe this will not be required as I finish this PR.